### PR TITLE
21.2 rebuild

### PR DESCRIPTION
**Pull request by aradi**
Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [X] Bumped the build number (if the version is unchanged)
* [ ] ~~Reset the build number to `0` (if the version changed)~~
* [X] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [X] Ensured the license file is being packaged.

Contains all changes planed for the next rebuild.

Closes #21
Closes #22

### CODE REVIEW

**Review by bhourahine — APPROVED** [review]
Seems ready to go.
Any specific reason for the .tar.xz instead of .tgz?

**Comment by aradi** [comment]
Not ready yet, I just had to activate to trigger the CI. :smile: 

As for the URL. In order to generate the proper header information within DFTB+, I take the release bundle (containing all submodule sources and a RELEASE file). That I usually compress with xz for the smaller size...

**Comment by aradi** [comment]
Now, it should be indeed ready for review :wink: 

After testing the binary more extensively, it seems that only the `solver/Si32C32_LS_OMM` example fails, yielding a strange error message in `MatrixSwitch.err` about a matrix C being complex... Not sure, what to do with that, I think we'll just live with it for the moment.

**Comment by aradi** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -32,6 +32,9 @@ build:
     - {{ name }} * {{ mpi_prefix }}_*
   {% endif %}
 
+  script_env:
+    - DFTBPLUS_TEST_BUILD="{{ build_platform == target_platform }}"
+
```
Comment (awvwgk) [inline recipe/meta.yaml]:
```suggestion
```

File: recipe/build.sh [hunk 2]
```diff
@@ -44,3 +44,45 @@ FFLAGS="-fno-backtrace" cmake "${cmake_options[@]}"
 ninja all install
 
 popd
+
+
+if [ "${DFTBPLUS_TEST_BUILD}" != "True" ]; then
+  exit 0
+fi
```
Comment (awvwgk) [inline recipe/build.sh]:
```suggestion
if [[ "${CONDA_BUILD_CROSS_COMPILATION:-0}" == "1" ]]; then
  exit 0
fi
```

**Review by awvwgk — APPROVED** [review]

File: recipe/build.sh (hunk 2 above)
Comment (aradi) [reply recipe/build.sh]: Very elegant, but how the hell did you know about this variable? :smile:  On https://docs.conda.io/projects/conda-build/en/latest/user-guide/environment-variables.html I could not find any hint about it...

Comment (awvwgk) [reply recipe/build.sh]: This one was introduced with the OSX/Arm migrations (https://conda-forge.org/docs/orga/minutes/2020-08-12.html?highlight=conda_build_cross_compilation), it is provided by the CTNG compiler activation (https://github.com/conda-forge/ctng-compiler-activation-feedstock).

**Pull request merged**